lib/importer.py

```diff
@@ -883,44 +883,27 @@ def execImport(handle: int, options: dict):
                 log(f"cannot import {mediaType} items from unknown library section {librarySection}", xbmc.LOGWARNING)
                 continue
 
-            # get all matching items from the library section
+            # get all matching items from the library section and turn them into ListItems
             try:
-                plexSectionItems = section.search(libtype=plexLibType)
-                plexItems.extend(plexSectionItems)
+                plexItems = section.search(libtype=plexLibType, container_size=ITEM_REQUEST_LIMIT)
             except plexapi.exceptions.BadRequest as e:
-                log(
-                    f"failed to retrieve {mediaType} items from {mediaProvider2str(mediaProvider)}: {e}",
-                    xbmc.LOGWARNING
-                )
+                log(f"failed to fetch {mediaType} items from {mediaProvider2str(mediaProvider)}: {e}", xbmc.LOGINFO)
                 return
 
-        # parse all items
-        items = []
-        itemsProgressTotal = len(plexItems)
-        for itemsProgress, plexItem in enumerate(plexItems):
-            if xbmcmediaimport.shouldCancel(handle, itemsProgress, itemsProgressTotal):
-                return
+            plexItemsProgressTotal = len(plexItems)
+            for plexItemsProgress, plexItem in enumerate(plexItems):
+                if xbmcmediaimport.shouldCancel(handle, plexItemsProgress, plexItemsProgressTotal):
+                    return
 
-            try:
                 item = Api.toFileItem(plexServer, plexItem, mediaType, plexLibType)
                 if not item:
                     continue
 
-                items.append(item)
-
-            except plexapi.exceptions.BadRequest as e:
```

Comment:
I didnt' see any code in or used by Api.toFileItem that makes calls back out to Plex that could raise BadRequest.

It is still up around the .search calls

Reply:
TBH I'm not sure. The comment below seems to indicate that the issue is with `Api.convertDateTimeToDbDateTime` but that sounds odd.

Reply:
That comment doesn't seem to describe a possible situation, but I'll just throw the try/catch back around it to be safe.

Reply:
This was added by @enen92 in 6337f684b0301c5ee5a6a66e88a5bac643797922. Maybe he can provide some details on this question.

Reply:
Yeah this was added because a single item (broken cache in your Plex server) could break the whole import process when you request details for the item. When this happened (it was Ralph breaks the internet in one of the test servers) this was the exception that was raised (result of 404) when querying the movie id.

Reply:
So if Api.tofileitem() still results in a call to the server, it's better to leave the exception there. The failure is rare but it can happen when you have a huge library; having it crash in those cases is not good as the import process takes a long time (at least on the first import)